idaplugin/rematch/actions/match.py

```diff
@@ -34,9 +34,17 @@ def perform(self):
       i, offset = self.function_gen.next()
 
       func = instances.FunctionInstance(self.file_id, offset)
-      network.query("POST", "collab/instances/", params=func.serialize(),
-                    json=True)
-
+      try:
+        network.query("POST", "collab/instances/", params=func.serialize(),
+                      json=True)
+      except exceptions.QueryException as e:
+        data = e.response()
+        if "Invalid pk" in data['file'][0]:
```

Comment:
I'm not sure what's the exact error, but can this be more generic? Are you sure the `[0]` here is appropriate?